crates/matrix-sdk-indexeddb/src/state_store/migrations.rs

```diff
@@ -422,18 +455,20 @@ async fn migrate_to_v4(
         data.insert(keys::KV, values);
     }
 
-    Ok(OngoingMigration {
+    db.close();
+
+    let migration = OngoingMigration {
         drop_stores: [old_keys::SYNC_TOKEN, old_keys::SESSION].into_iter().collect(),
         create_stores: [keys::KV].into_iter().collect(),
         data,
-    })
+    };
+    apply_migration(name, 4, migration).await
```

Comment:
Here it's explicitly closed and then implicitly reopend in `apply_migration`; could this function take an `Option<OpenDbRequest>` parameter that is filled when the database had been opened, using `unwrap_or_else` to open it if it wasn't opened in the first place?

Reply:
We _have_ to drop the connection before calling `apply_migration`. We can only create/remove stores in `onupgradeneeded`, which in turn is only called when opening the DB with a version greater than the current one.

Note that this code is confusing mainly because of the indexeddb_db_futures crate that doesn't allow us to read or change data in the stores inside `onupgradeneeded`. If we could, we would only need to open the DB once.